### PR TITLE
Build doc components separately

### PR DESCRIPTION
**Pull request by duck-rh**
This is needed to publish plugin documentation separately from the core.

Closes #7229

### CODE REVIEW

File: playbooks/deploy.yml [hunk 1]
```diff
@@ -60,41 +60,58 @@
...
-    # rsync is already installed for the web builder sync
-    - name: Install doc builder's SSH keys
-      authorized_key:
-          key: "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDNfEAXdhT7mIf4d+XOWvE4hw5uduVyqc1N3ttqHV/5SM37XWHvln9D0IzMtDllXrmvlFuUA8sUCnc3zw3s+zIFh0ln+nhEQAo3StvchmlzmYJD1VyS1e8Pfd9xnD8DOBQNdy3aq4ZVrLtCAYndfX1Y5sfqlgN35OQ+7frjKaJXQWVHuLDGTxWR/y82EOT5JyXPyZd6B9VsQRPmB5oK1eSqMNQikuXJITDRxK6olfIdLzze85gV0aRJYo6GD5Aj68SciIsR95y//afwOQZr8m7OjygTsxLTlzsoRVR2N4uTyOe2i2szqy9kxUnvTiTVDULWicQZLBmv7S1pjaytKJnj pulp-doc-builder"
-          user: "{{ doc_user }}"
+        # TODO: enable when pulpcore is properly synced
+        #redirects:
+        #- src: '^/$'
+        #  target: '/pulpcore'
+        #  match: True
```
Comment (dkliban) [inline playbooks/deploy.yml]: Can we also add a redirect rule that redirects a requested URL that does not start with "/pulp_" or "/pulpcore" to "/pulpcore/{$url}" ?

Comment (duck-rh) [reply playbooks/deploy.yml]: yes, will do that on Monday (it's late here ;-P)

Comment (dkliban) [reply playbooks/deploy.yml]: Thank you! Have a good weekend!

Comment (dkliban) [reply playbooks/deploy.yml]: @duck-rh How's this going?

Comment (duck-rh) [reply playbooks/deploy.yml]: @dkliban I made the changed and it seems to work fine.

File: data/web/doc_old_layout_redirection.conf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+RewriteEngine On
+RewriteCond %{REQUEST_URI} !^/pulpcore
+RewriteCond %{REQUEST_URI} !^/pulp_
```
Comment (dkliban) [inline data/web/doc_old_layout_redirection.conf:4]: can you also add !^/en/2 to the rules?

Comment (dkliban) [reply data/web/doc_old_layout_redirection.conf:4]: That's for Pulp 2 docs that are not part of pulpcore.

Comment (dkliban) [reply data/web/doc_old_layout_redirection.conf:4]: @duck-rh ^ ... The redirects are working great. That's the only additional change we need before merging this.

Comment (duck-rh) [reply data/web/doc_old_layout_redirection.conf:4]: @dkliban added

**Pull request merged**